### PR TITLE
Fix formatting of field override with constraint

### LINKED ISSUE #1537: Bug: Invalid AST on field override with constraint
```ocaml
let _ = {< x = (x : t) >}
```

This formats to `let _ = {<x : t>}`, which fails.
This case is explicitly handled (`Pexp_override`) but I think no syntax exist for this.

### PR DESCRIPTION
**Pull request by gpetiot**
Fix #1537, no diff with test_branch.sh

### CODE REVIEW

**Review by Julow — APPROVED** [review]
Looks good !

**Review by Julow — CHANGES_REQUESTED** [review]
I approved too fast. Here is an input that still fails:
```ocaml
let _ = {< x = x [@a] >}
```
The attribute is removed too.

**Comment by gpetiot** [comment]
Good catch! I fixed the simplified case and added some tests.

File: test/passing/override.ml.ref [hunk 1]
```diff
@@ -0,0 +1,9 @@
+let _ = {<x = (x : t)>}
+
+let _ = {<x = ((x [@a]) : (t[@b])) [@c]>}
+
+let _ = {<x>}
+
+let _ = {<x = (x [@a])>}
```
Comment (Julow) [inline test/passing/override.ml.ref:7]: These parentheses are optional but I think it's fine to keep them for readability. We need to keep track of this kind of decisions we take somewhere.

**Review by Julow — APPROVED** [review]
Let's merge !

**Pull request merged**